### PR TITLE
Add basic test cases for most of EVM->Ewasm translator

### PR DESCRIPTION
**Pull request by axic**
In preparation for merging #7766.

### CODE REVIEW

File: test/libyul/ewasmTranslationTests/balance.yul [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  sstore(0, balance(address()))
+  sstore(1, balance(0))
+}
+// ----
+// Trace:
+// Memory dump:
+//      0: 0000000000000000000000000000000011111111000000000000000000000000
```
Comment (chriseth) [inline test/libyul/ewasmTranslationTests/balance.yul:8]: looks wrong.

Comment (chriseth) [reply test/libyul/ewasmTranslationTests/balance.yul:8]: but it may be just the interpreter.

File: test/libyul/ewasmTranslationTests/invalid.yul [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  invalid()
+  sstore(0, 5678)
+  mstore(1, 0xff)
+}
+// ----
+// Trace:
+// Memory dump:
+//     20: 00000000000000000000000000000000000000000000000000000000000004d2
+// Storage dump:
+//   0000000000000000000000000000000000000000000000000000000000000000: 00000000000000000000000000000000000000000000000000000000000004d2
```
Comment (chriseth) [inline test/libyul/ewasmTranslationTests/invalid.yul]: Does the interpreter correctly evaluate `invalid()`?

**Review by chriseth — CHANGES_REQUESTED** [review]
Can be merged as an initial baseline.

File: test/libyul/ewasmTranslationTests/balance.yul (hunk 1 above)
Comment (axic) [reply test/libyul/ewasmTranslationTests/balance.yul:8]: `balance` is not implemented yet by the translator

File: test/libyul/ewasmTranslationTests/invalid.yul (hunk 2 above)
Comment (axic) [reply test/libyul/ewasmTranslationTests/invalid.yul]: I've realised it too that with decimals it is hard to follow. Changed to hex and it confirms to be working.

**Comment by axic** [comment]
> Can be merged as an initial baseline.

@chriseth I'd prefer that too. Created an issue, #7956, to track it.

**Review by chriseth — APPROVED** [review]
